### PR TITLE
License cannot be applied to the RPP project[#OSF-5389]

### PR DESCRIPTION
**Pull request by chennan47**
## Purpose

Excluded deleted and private node from update search when try to save nodes for node_license update. Can be a solution to the problem of https://openscience.atlassian.net/browse/OSF-5389. Even if it does not solve this ticket, deleted nodes and private nodes shouldn't be included in updates for node_license search update. And we can still use https://github.com/CenterForOpenScience/osf.io/pull/6054 to add license to rpp as a ultimate solution.

## Ticket

… nodes for node_license update[#OSF-5389]
https://openscience.atlassian.net/browse/OSF-5389.

### CODE REVIEW

**Comment by brianjgeiger** [comment]
This looks good once tests pass. 🎉 🐧 🎉

File: website/project/model.py [hunk 1]
```diff
@@ -1689,10 +1689,14 @@ def save(self, *args, **kwargs):
...
             children = [c for c in self.get_descendants_recursive(
                 include=lambda n: n.node_license is None
-            )]
+            ) if c.is_public and not c.is_deleted]
             # this returns generator, that would get unspooled anyways
             if children:
-                Node.bulk_update_search(children)
+                if len(children) > 100:
+                    Node.bulk_update_search(children[:99])
+                    Node.bulk_update_search(children[99:])
```
Comment (brianjgeiger) [inline website/project/model.py]: We won't run into the problem for a while, but since we're solving this anyways, let's make this handle an arbitrary number of nodes. Perhaps a loop?

**Comment by brianjgeiger** [comment]
Almost there. Pass done. 🐧

File: website/project/model.py [hunk 2]
```diff
@@ -1689,10 +1689,20 @@ def save(self, *args, **kwargs):
...
+                if count == 0:
+                    Node.bulk_update_search(children)
+                else:
+                    start = 0
+                    end = 100
+                    for i in range(1, count + 1):
+                        Node.bulk_update_search(children[start:end])
+                        start = end
+                        end = 100 * (i + 1)
+                    Node.bulk_update_search(children[(100 * count):])
```
Comment (brianjgeiger) [inline website/project/model.py]: This looks like it might hit a problem on an edge case, so we'd either want to break it out into a function and unit test that function, or we can possibly simplify with something like:

```py
while len(children):
    batch = children[:99]
    children = children[99:]
    Node.bulk_update_search(batch)
```

**Comment by brianjgeiger** [comment]
Pass done. Mostly me being picky because it's in the Node's save method. 🐧

**Comment by brianjgeiger** [comment]
Looks good. 🎉 🐧 🎉  This is approved once tests pass.
